### PR TITLE
Add CI job to publish docker images description

### PR DESCRIPTION
**Pull request by sergejparity**
Added CI job to publish Docker image description to hub.docker.com for [faucet-bot](https://hub.docker.com/r/paritytech/faucet-bot) and [faucet-server](https://hub.docker.com/r/paritytech/faucet-server)

Description fetched from a files in a repository (located side by side with Dockerfiles and having a suffix `README.md`) and as soon they are updated and merged into the `main` branch, its content will get published to relevant image description at hub.docker.com

Related https://github.com/paritytech/ci_cd/issues/741

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -178,3 +196,17 @@ westend:
...
+  extends:                         .publish-docker-image-description
+  variables:
+    SHORT_DESCRIPTION:             "Faucet bot component"
+    COMPONENT:                     bot
+
+publish-server-docker-image-description:
+  extends:                         .publish-docker-image-description
+  variables:
+    SHORT_DESCRIPTION:             "Faucet server component"
+    COMPONENT:                     server
```
Comment (Bullrich) [inline .gitlab-ci.yml:212]: @rzadp would this collide with #221?

Comment (sergejparity) [reply .gitlab-ci.yml:212]: @Bullrich you're right changes introduced by @rzadp changes the game a bit :)
So I'll put this one on hold till #221 get merged. And then will adjust accordingly.
Good catch. Thank you.

Comment (sergejparity) [reply .gitlab-ci.yml]: Updated to reflect changes from #221

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -157,3 +158,22 @@ westend:
   <<:                              *deploy-prod-refs
   <<:                              *kubernetes-env
   <<:                              *deploy-k8s
+
+# stage:                           publish-docker-description
```
Comment (rzadp) [inline .gitlab-ci.yml:162]: Leftover?

**Review by rzadp — APPROVED** [review]

**Pull request merged**